packages/ra-core/src/sideEffect/useRedirect.ts

```diff
@@ -37,12 +37,14 @@ const useRedirect = () => {
             basePath: string = '',
             id?: Identifier,
             data?: Partial<Record>
-        ) =>
-            redirectTo
-                ? dispatch(
-                      push(resolveRedirectTo(redirectTo, basePath, id, data))
-                  )
-                : dispatch(reset('record-form')), // explicit no redirection, reset the form
+        ) => {
+            if (!redirectTo) {
+                dispatch(refreshView());
```

Comment:
I'm surprised that a simple refresh is enough to empty a create form. Do you confirm it works?

Reply:
1. What is a simple refresh ?
2. Yes and we have e2e tests for all redirection cases (see `PostCreate` which redirect to `edit`, `show`, `create`)